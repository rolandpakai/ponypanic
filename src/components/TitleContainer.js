import { Fragment, useContext } from 'react'; 
import Box from '@mui/material/Box';
import Container from '@mui/material/Container';
import Stack from "@mui/material/Stack";

import { ThemeContext } from '../contexts/ThemeContext';
import { NewGameContext } from '../contexts/NewGameContext';
import Button from './Button';

const TitleContainer = () => {
  const { theme } = useContext(ThemeContext);
  const { setNewGame } = useContext(NewGameContext);

  const onClickHandler = () => {
    setNewGame(true);
  }
  
  return (
    <Fragment>
      <Box>
        <Container 
          maxWidth="md" 
        >
          <Box 
            justifyContent="center" 
            sx={{  
              height: '550px', 
              padding: '2rem',
              backgroundPosition: 'center', 
              backgroundRepeat: 'no-repeat', 
              backgroundImage: `url('./themes/${theme}/title-screen.png')` }}
          />
        </Container>
        <Container 
          maxWidth="xs" 
        >
          <Stack 
            maxWidth="xs"
            direction="row" 
            justifyContent="center" 
          >
            <Button onClick={onClickHandler}>
              {"NEW GAME"} 
            </Button>
          </Stack>
        </Container>
      </Box>
  </Fragment>
  )
}

export default TitleContainer;