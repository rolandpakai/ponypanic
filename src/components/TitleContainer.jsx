import React, { useContext } from "react";
import Box from "@mui/material/Box";
import Container from "@mui/material/Container";
import Stack from "@mui/material/Stack";

import { ThemeContext } from "../contexts/ThemeContext";
import { NewGameContext } from "../contexts/NewGameContext";
import Button from "./Button";

const TitleContainer = () => {
  const { theme } = useContext(ThemeContext);
  const { setNewGame } = useContext(NewGameContext);

  const onClickHandler = () => {
    setNewGame(true);
  };

  return (
    <Box>
      <Container maxWidth="md">
        <Box
          justifyContent="center"
          sx={{
            height: "550px",
            padding: "2rem",
            backgroundPosition: "center",
            backgroundRepeat: "no-repeat",
            backgroundImage: `url('./themes/${theme}/title-screen.png')`,
          }}
        />
      </Container>
      <Container maxWidth="xs">
        <Stack direction="row" justifyContent="center" maxWidth="xs">
          <Button onClick={onClickHandler}>NEW GAME</Button>
        </Stack>
      </Container>
    </Box>
  );
};

export default TitleContainer;
