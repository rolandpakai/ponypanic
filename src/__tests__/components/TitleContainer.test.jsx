import React from "react";
import renderer from "react-test-renderer";
import TitleContainer from "../../components/TitleContainer";

describe("TitleContainer Component", () => {
  test("TitleContainer renders", () => {
    const container = renderer.create(<TitleContainer />).toJSON();

    expect(container).toMatchSnapshot();
  });
});
