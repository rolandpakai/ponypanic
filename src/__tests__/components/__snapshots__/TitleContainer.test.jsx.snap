// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleContainer Component TitleContainer renders 1`] = `
<div
  className="MuiBox-root css-0"
>
  <div
    className="MuiContainer-root MuiContainer-maxWidthMd css-xdmu94-MuiContainer-root"
  >
    <div
      className="MuiBox-root css-1js63x3"
    />
  </div>
  <div
    className="MuiContainer-root MuiContainer-maxWidthXs css-nzinn5-MuiContainer-root"
  >
    <div
      className="css-vankme-MuiStack-root"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-fullWidth MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-fullWidth css-dat04h-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
        disabled={false}
        id=":r0:"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        NEW GAME
      </button>
    </div>
  </div>
</div>
`;
